Lib/test/test_math.py

```diff
@@ -1756,6 +1756,10 @@ def test_prod(self):
         with self.assertRaises(TypeError):
             prod([10, 20], [30, 40])     # start is a keyword-only argument
 
+        self.assertEqual(prod([0, 1, 2, 3]), 0)
+        self.assertEqual(prod([1, 0, 2, 3]), 0)
```

Comment:
`[1, 2, 3, 0]` was already working fine without this fix so is it worth adding this as a test case? I have less knowledge with C so please ignore if this is not needed.

```
self.assertEqual(prod([1, 2, 3, 0]), 0)
```

Reply:
I have added this test and removed the `range` base one. Even if that case worked fine because it takes a different code path, I like to have the 0 in the start/middle/end. 

Thanks for the catch!